app/rule/[ruleId]/embedded/page.tsx

```diff
@@ -1,14 +1,20 @@
+import getRuleDataForVersion from "@/app/hooks/getRuleDataForVersion";
 import SimulationViewer from "../../../components/SimulationViewer";
-import { getRuleDataById, getScenariosByFilename } from "../../../utils/api";
+import { getScenariosByFilename } from "../../../utils/api";
 import { Scenario } from "@/app/types/scenario";
+import { RULE_VERSION } from "@/app/constants/ruleVersion";
 
 export default async function Rule({ params: { ruleId } }: { params: { ruleId: string } }) {
-  const { _id, goRulesJSONFilename } = await getRuleDataById(ruleId);
-  const scenarios: Scenario[] = await getScenariosByFilename(goRulesJSONFilename);
+  // Get rule details and json content for the rule id
+  const { ruleInfo, ruleContent } = await getRuleDataForVersion(ruleId, RULE_VERSION.published);
```

Comment:
Currently see a 'rule not found' page when creating a draft of a brand new rule, saving, and navigating back to the home page and to the new rule - as there is no published version for this new rule. Could we update so that it tries for const { ruleInfo, ruleContent } = await getRuleDataForVersion(ruleId, RULE_VERSION.draft) in the if statement before returning 'rule not found'?

Reply:
Yeah, I definitely want to improve this experience. I was thinking of adding some sort of `isPublished` flag to the db in future to help with this. But your suggestion here could be a good middle ground. Maybe just straight up redirecting to the draft with a message.